packages/api/src/modules/associations/associations.service.ts

```diff
@@ -139,16 +139,19 @@ export class AssociationsService {
         return Object.values(providers).filter(p => this.isAssociationsProvider(p)) as AssociationsProvider[];
     }
 
-    async isSirenFromAsso(siren: Siren): Promise<boolean> {
-        if (await dataGouvService.sirenIsEntreprise(siren)) return false;
+    async isIdentifierFromAsso(identifier: StructureIdentifiers, knownIdType?: StructureIdentifiersEnum) {
+        const idType = knownIdType || getIdentifierType(identifier);
+        if (!idType) throw new StructureIdentifiersError();
 
-        // what follows will be useless when #554 is done (then maybe the helper will be redundant)
-        if (await rnaSirenService.getRna(siren)) return true;
+        if (idType === StructureIdentifiersEnum.rna) return true;
+        // from there it is either siren or siret
+        return !(await dataGouvService.sirenIsEntreprise(siretToSiren(identifier)));
+    }
 
-        const asso = await apiAssoService.findAssociationBySiren(siren);
-        if (!asso?.categorie_juridique?.[0]?.value) return false;
-        return LEGAL_CATEGORIES_ACCEPTED.includes(asso.categorie_juridique[0].value);
-    } // TODO tests
+    async validateIdentifierFromAsso(identifier: StructureIdentifiers, knownIdType?: StructureIdentifiersEnum) {
```

Comment:
Je nommerai simplement `validateIdentifier`; on est dans le `association.service` selon moi ça fait sens

Reply:
Je suis pas convaincue : le sens du nom c'est "valide que l'identifiant vient bien d'une association" mais il ne valide pas que l'identifiant est bien formé par ex. Pour moi, `validateIdentifier` voudrait dire que ça valide tous les aspects de l'identifiant. Mais peut-être qu'il peut y avoir un meilleur nom

Reply:
Est un truc genre isIdentifierFromAsso ?

Reply:
bah c'est le nom de la fonction appelée à l'intérieur ! Mais un `isIdentifierValid` qui appel un `isIdentifierFromAsso` et peut-être demain un `isIdentifierXXX` ça fait sens

Reply:
Mais on a des déjà des validateurs qui vérifient qu'un identifiant est un rna ou un siren mais ç'a besoin d'être appelé à des endroits un peu différents de mémoire, donc pas forcément facile à rejoindre
Le `validate` appelle juste le `is` mais lève une exception si `is` renvoie `false`

Reply:
si on se dit que les méthodes `validateXXX` ne font qu'appeler un validateur et levé une exception alors on a un nouveau pattern décrit et ça me va

Reply:
Je suis pas sûre que ce soit déjà le cas partout ailleurs mais c'était mon raisonnement oui. Je sais pas si ton "si" était un élément d'explication et la conclusion c'est "ça me va" ou s'il faut discuter du "si"